File: LeetCode/26-Remove-Duplicates-from-Sorted-Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let n = nums.length;

        if (n == 0 || n == 1)
            return n;

        let j = 0;
     
        for (let i = 0; i < n - 1; i++)
            if (nums[i] != nums[i + 1])
                nums[j++] = nums[i];
     
        nums[j++] = nums[n - 1];
     
        return j;
};

File: 26-Remove-Duplicates-from-Sorted-Array.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) {


        //idea: remove the duplicate from array

        //approch: 
        //1-check duplicate, if duplicate remove, else add to new array
        //2- print new array
        
        int expectedNums = 0;

        for(int i =1; i < nums.Length; i++){
            if(nums[expectedNums] != nums[i]){
                nums[++expectedNums] = nums[i];
            }
        }
        return expectedNums+1;
        
        
    }
}